### PR TITLE
[FLINK-8151][table]Map equality check to use entrySet equality

### PR DESCRIPTION
**Pull request by walterddr**
Remove map type equality support as unlike java array, there's no element-wise equality check on Map type. 

Also small clean up on map value constructor code.

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala [hunk 1]
```diff
@@ -188,13 +188,6 @@ object ScalarOperators {
         (leftTerm, rightTerm) => s"java.util.Arrays.equals($leftTerm, $rightTerm)"
       }
     }
-    // map types
-    else if (isMap(left.resultType) &&
-      left.resultType.getTypeClass == right.resultType.getTypeClass) {
-      generateOperatorIfNotNull(nullCheck, BOOLEAN_TYPE_INFO, left, right) {
-        (leftTerm, rightTerm) => s"java.util.Map.equals($leftTerm, $rightTerm)"
```
Comment (twalthr) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala]: Do we really need to remove the entire functionality? Wouldn't be `$leftTerm.equals($rightTerm)` be enough?

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/MapTypeTest.scala [hunk 2]
```diff
@@ -139,13 +139,6 @@ class MapTypeTest extends MapTypeTestBase {
   @Test
   def testMapOperations(): Unit = {
 
-    // comparison
-    testAllApis(
```
Comment (twalthr) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/MapTypeTest.scala]: Why did this test ever worked?

Comment (walterddr) [reply flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/MapTypeTest.scala]: That was my bad, I was only testing the invalid equality case ( e.g. `MAP[STRING, INT] = MAP[STRING,STRING]` should throw ValidationException), I did not put in the valid equality test :-(

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala (hunk 1 above)
Comment (walterddr) [reply flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/ScalarOperators.scala]: I tried to use this but this is an exact object match, which is very different from `Array.equals`, which is indexed element match. so there's a problem here: 
1. an object referred on both the left and right hand side originally might become different object depending on the ser/deser (and they still represent the same Map); 
2. if two Map objects with exact same key and value pairs, they should be considered equal, however they are not equal under object match;
3. it is also a debate on whether `Map` should be considered as ordered or unordered map.

I think this worth another PR to address.

**Comment by fhueske** [comment]
`Map` does not provide an implementation for `equals`. However, `java.util.AbstractMap` does which is the base class for the commonly used Java maps including `HashMap`. Maybe we can add a case for `AbstractMap`?

**Comment by walterddr** [comment]
Ah. I see. `AbstractMap.equals` look very much similar to `Arrays.equals`. I will add it in and some more testings. thanks @fhueske for the great tips

**Comment by walterddr** [comment]
@twalthr I think my previous unit test method was not correct. Since TableAPI always use `java.util.HashMap` per flink documentation. I think it is safe to directly use `$leftTerm.equals($rightTerm)` and it should always be using `java.util.AbstractMap.equals` implementation. Do you think I should add extra check or do you think the `isMap(resultType)` should be suffice?

**Comment by twalthr** [comment]
@walterddr I think the current solution is sufficient for now. I will merge this...

**Pull request closed without merging**